Sprint 12

Grifon-com opened the pull request:
Hi. 
Все error пока кидаю в print, может лучше assetionFailure или по ситуации?
Легкой проверки))

Grifon-com committed:
создал сетевой слой для получения фотографий

Grifon-com committed:
1. добавил в функциональность лайков,
2. добавил полноэкранный просмотр фотографии.

Grifon-com committed:
Добавил клас Clear для очистки все куки из хранилища и удаления токена

Grifon-com committed:
Добавил класс для работы с датой

Grifon-com committed:
Добавил показ индикатора загрузки таблицы

Grifon-com committed:
рефакторинг

Grifon-com committed:
Рефакторинг

Grifon-com committed:
Добавил выход из профиля по кнопке "Да" алеута

Grifon-com committed:
Рефакторинг

Grifon-com committed:
Рефакторинг

Grifon-com committed:
рефакторинг

Grifon-com committed:
рефакторинг

Grifon-com committed:
Рефакторинг

Grifon-com committed:
Установил SplashView рутовым  после нажатия кнопки "ДА" алерта

amardanov commented on ImageFeed/AppDelegate.swift:
```diff
@@ -6,11 +6,11 @@
 //
 
 import UIKit
+import ProgressHUD
```
Не увидел использования этой библиотеки в данном файле, поэтому советую удалить этот `import` отсюда

amardanov commented on ImageFeed/Constants/Constants.swift:
```diff
@@ -5,7 +5,7 @@
-import Foundation
+import UIKit
 
 enum ConstantsUnSplash {
```
Советую все константы убрать в один `struct`, потому что создавать под каждый тип новый файл/енам затратно бывает)

amardanov commented on ImageFeed/Services/Clean.swift:
```diff
@@ -0,0 +1,35 @@
+    func cleanToken()
+}
+
+final class Clean: CleanProtocol {    
```
лучше назвать файл и класс как-то по типу `CleanManager` или `CleanService`. Так будет более понятно за что отвечает этот класс

amardanov commented on ImageFeed/Services/ImagesListService.swift:
```diff
@@ -0,0 +1,223 @@
+
+final class ImagesListService: ImagesListServiceProtocol {
+    private static let sizeDefault = (heigt: 0, width: 0)
+    private static let userInfoKey = "listPhoto"
```
Лучше все такие константы убирать в структуру/enum внутри сервиса, чтобы унифицировать их хранение и сделать их использование более удобным и понятным

Grifon-com replied:
Спасибо за рекомендации

amardanov commented on ImageFeed/Services/ImagesListService.swift:
```diff
@@ -0,0 +1,223 @@
+                completion(.failure(error))
+            }
+        }
+        if isLike {
```
Можно сделать в одну строчку
`isLike ? self.deleteLike(photoId: photoId, completion: completionHandler) : self.putLike(photoId: photoId, completion: completionHandler)`

amardanov commented on ImageFeed/Services/ImagesListService.swift:
```diff
@@ -0,0 +1,223 @@
+        guard let token = OAuth2TokenKeychainStorage().getToken() else { throw KeychainError.errorStorageToken}
+        let bearerToken = "\(ConstantsUnSplash.bearer) \(token)"
+        
+        if isLike {
```
здесь тоже можно в одну строку 
`return URLRequest.makeHTTPRequestForToken(url: url, bearerToken: bearerToken, forHTTPHeaderField: ConstantsUnSplash.hTTPHeaderField, httpMethod: isLike? ConstantsUnSplash.postHTTPMethod : ConstantsUnSplash.deleteHTTPMethod)`

amardanov commented on ImageFeed/TabBar/ImageList/ImagesListViewController.swift:
```diff
@@ -6,25 +6,28 @@
-    private static let edgeInsetsTableView = (top: 12, left: 0, bottom: 12, right: 0)
-    private static let edgeInsetsCellView = (top: 4, left: 16, bottom: 4, right: 16)
+    private static let placeholder = "placeholderCell"
+    private static let contentInsetsTableView = UIEdgeInsets(top: 12, left: 0, bottom: 12, right: 0)
```
более понятно будет звучать и выглядеть `tableViewContentInsets`. Переименуй все такие кексы пожалуйста

amardanov commented on ImageFeed/TabBar/ImageList/ImagesListViewController.swift:
```diff
@@ -110,3 +200,32 @@ private extension ImagesListViewController {
 }
+ 
+extension ImagesListViewController: ImageListCellDelegate {
+    internal func imageListCellDidTapLike(_ cell: ImagesListCell) {
```
`internal` можно не писать, так как в Свифте все методы/поля по умолчанию `internal`

amardanov commented on ImageFeed/TabBar/ProfileViewController/ProfileViewController.swift:
```diff
@@ -14,12 +14,19 @@ final class ProfileViewController: UIViewController {
-    private static let imagePlaceholderName = "placeholder"
+    private static let imagePlaceholderName = "placeholderAvatar"
+    
+    private static let titleAlert = "Пока, пока!"
```
Тоже лучше в константы внутри файла вынести такие поля

amardanov commented on ImageFeed/TabBar/SingleImageViewController/SingleImageViewController.swift:
```diff
@@ -120,7 +144,7 @@ private extension SingleImageViewController {
         view.addSubview(sharedButton)
         
-        imageView.image = image
+//        imageView.image = image
```
комментарии такие лучше не оставлять с кодом

Grifon-com replied:
Спасибо за все рекомендации)

Grifon-com committed:
1.  устранил возможное возникновение гонки при запросе на like
2. вынес static let классов в отдельные структуры
3. сделал поле createAt структуры Photo опциональным

Grifon-com committed:
Добавил локаль в DateFormatter
